services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/links/LinkSpec.groovy

```diff
@@ -0,0 +1,98 @@
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
+
+class LinkSpec extends BaseSpecification {
+
+    @Autowired
+    TopologyDefinition topology
+    @Autowired
+    FlowHelper flowHelper
+    @Autowired
+    PathHelper pathHelper
+    @Autowired
+    NorthboundService northboundService
+
+    def "Get all flows going through a particular existing link"() {
+        given: "Two active not neighboring switches"
+        def switches = topology.getActiveSwitches()
+        def allLinks = northboundService.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            allLinks.every { link ->
+                def switchIds = link.path*.switchId
+                !(switchIds.contains(src.dpId) && switchIds.contains(dst.dpId))
+            }
+        }
+        assert srcSwitch && dstSwitch
+
+        and: "Forward flow from source switch to destination switch"
+        def flow1 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow1.maximumBandwidth = 1000
+        flow1 = northboundService.addFlow(flow1)
+        assert Wrappers.wait(5) { northboundService.getFlowStatus(flow1.id).status == FlowState.UP }
+
+        and: "Reverse flow from destination switch to source switch"
+        def flow2 = flowHelper.randomFlow(dstSwitch, srcSwitch)
+        flow2.maximumBandwidth = 1000
+        flow2 = northboundService.addFlow(flow2)
+        assert Wrappers.wait(5) { northboundService.getFlowStatus(flow2.id).status == FlowState.UP }
+
+        and: "Forward flow from source switch to some 'internal' switch"
+        def internalSwitch = switches.find {
+            it.dpId == northboundService.getFlowPath(flow1.id).forwardPath[1].switchId
+        }
+
+        def flow3 = flowHelper.randomFlow(srcSwitch, internalSwitch)
+        flow3.maximumBandwidth = 1000
+        flow3 = northboundService.addFlow(flow3)
+        assert Wrappers.wait(5) { northboundService.getFlowStatus(flow3.id).status == FlowState.UP }
+
+        and: "Reverse flow from 'internal' switch to source switch"
+        def flow4 = flowHelper.randomFlow(internalSwitch, srcSwitch)
+        flow4.maximumBandwidth = 1000
+        flow4 = northboundService.addFlow(flow4)
+        assert Wrappers.wait(5) { northboundService.getFlowStatus(flow4.id).status == FlowState.UP }
+
+        when: "Get all flows going through the link from source switch to 'internal' switch"
+        def isl = pathHelper.getInvolvedIsls(PathHelper.convert(northboundService.getFlowPath(flow3.id))).first()
+        def linkFlows = northboundService.getLinkFlows(isl.srcSwitch.dpId, isl.srcPort, isl.dstSwitch.dpId, isl.dstPort)
+
+        then: "All created flows are in response list"
+        [flow1, flow2, flow3, flow4].every { it in linkFlows }
```

Comment:
check result for flow1/2 unique link, it should not contain flow3/4

Reply:
Done.